text/0000-project-descriptor.md

```diff
@@ -0,0 +1,324 @@
+
+```toml
+[project]
+include = [
+    "cmd/",
+    "go.mod",
+    "go.sum",
+    "*.go"
+]
+```
+
+A list of files to exclude from the build (while including everything else)
+
+```toml
+[project]
+exclude = [
+    "spec/"
+]
+```
+
+The `.gitignore` pattern is used in both cases. The `exclude` and `include` keys are mutually exclusive, and if both are present the Lifecycle will error out. These lists apply to both buildpacks built with `pack create-package` and apps built with `pack build`.
+
+Any files that are excluded (either via `include` or `exclude`) will be excluded before the build (i.e. not only exluded from the final image).
+
+## `[[images]]`
+
+Defines properties of the image(s) that are output from a build.
+
+```toml
+[[images]]
+ref = "<string>"
+suffix = "<string>"
+path = "."
+```
+
+* `ref` (string, optional): by default inherits from `[project.id]`. If set will be used as the tag of the OCI image produced
+* `suffix` (string, optional): If set will be used as the suffix to the tag of the OCI image produced
+* `path` (string, optional): by default uses the directory where this file lives
+
+## `[image.launch.env]` and `[image.build.env]`
+
+Used to set environment variables at launch time, for example:
+
+```toml
+[[images.launch.env]]
+name = "JAVA_OPTS"
+value = "-Xmx4g"
+```
+
+Or set environment variables at build time:
+
+```toml
+[[images.build.env]]
+name = "JAVA_OPTS"
+value = "-Xmx1g"
+```
+
+## `[[images.processes]]`
+
+Defines the default processes used to run an image.
+
+```toml
+[[images.processes]]
+name = "<process name>"
+cmd = "..."
+```
+
+This should mirror what is supported in `launch.toml`, which means that if `args` are added to support scratch images, they should be added here too.
+
+## `[[images.buildpacks]]`
+
+The images table may also contain an array of buildpacks. The schema for this table is:
+
+```toml
+[[images.buildpacks]]
+id = "<buildpack ID (required)>"
+version = "<buildpack version (optional default=latest)>"
+uri = "<url or path to the buildpack (optional default=urn:buildpack:<id>)"
+```
+
+This allows for each image to be built from a different group of buildpacks. There is an open question about how to simplify this interface if the buildpacks are the same for a number of a different `[[images]]`.
+
+## `[metadata]`
+
+This table includes a some defined keys, but additional keys are not validated. It can be used to add platform specific metadata. For example:
+
+```toml
+[metadata.heroku]
+pipeline = "foobar"
+```
+
+The defined keys are:
+
+```toml
+[metadata]
+authors = ["<string>"]
+documentation = "<url>"
+license = "<string>"
+source = "<url>"
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Given the elements described above, we seek to satisfy the following use cases:
+
+## Example: Basic app
+
+This is a complete `project.toml` for a simple app:
+
+```toml
+[project]
+id = "io.buildpacks.my-app"
+version = "0.1"
```

Comment:
Seems unusual to require a version to be checked into a git repo for project.toml to be valid. That breaks certain release strategies.